src/app/todo-list/services/bounty-catalog.service.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from '@angular/core';
+import { ItemType } from '@app/service/model';
+import { BehaviorSubject, combineLatest, Observable, of } from 'rxjs';
+import { filter, switchMap, takeUntil } from 'rxjs/operators';
+
+import { API_ROOT } from '../constants/constants';
+import { ApiResponse } from '../interfaces/api.interface';
+import { Character } from '../interfaces/player.interface';
+import {
+  Bounty,
+  InventoryItem,
+  SaleItem,
+  Vendor,
+  VendorResponse,
+  VendorSales,
+  VendorsSalesList,
+} from '../interfaces/vendor.interface';
+import { Destroyable } from '../util/destroyable';
+import { ContextService } from './context-service';
+import { DictionaryService } from './dictionary.service';
+import { HttpService } from './http.service';
+
+
+/**
+ * Provides a catalog of bounties
+ */
+@Injectable()
+export class BountyCatalogService extends Destroyable {
+
+  public bountyCatalog: BehaviorSubject<Bounty[]> = new BehaviorSubject(null);
+
+  private chars: Character[];
+  private uniqueBounties: { [key: string]: Bounty } = {};
+
+  constructor(
+    private http: HttpService,
+    private context: ContextService,
+    private dictionary: DictionaryService
+  ) {
+    super();
+    this.fetchBountiesForChars();
+  }
+
+  private fetchBountiesForChars() {
+    this.context.characters.pipe(
+      filter((x) => !!x),
+      switchMap((chars) => {
+        this.chars = chars;
+        const dataStream = this.createVendorsStream(chars);
+        return combineLatest(...dataStream);
+      }),
```

Comment:
Once the context service has the characters loaded, we can go fetch the vendor info. There's a TON of info that I don't really need, but I don't think I can request less data and still get the saleStatus for all bounties for all characters. Either way, this way is code-easy, but not as performant as possible, which I'm definitely fine with for MVP